odbc/src/api/get_info_tests.rs

```diff
@@ -0,0 +1,633 @@
+    expected_sql_return = $expected_sql_return:expr,
+    $(buffer_length = $buffer_length:expr,)?
+    $(expected_length = $expected_length:expr,)?
+    $(expected_value = $expected_value:expr,)?
+    $(actual_value_modifier = $actual_value_modifier:ident,)?
+    ) => {
+        #[test]
+        fn $func_name() {
+            unsafe {
+                let info_type = $info_type;
+
+                #[allow(unused_mut, unused_assignments)]
+                let mut conn =
+                    Connection::with_state(std::ptr::null_mut(), ConnectionState::Connected);
+                let mongo_handle: *mut _ = &mut MongoHandle::Connection(conn);
+
+                let value_ptr: *mut std::ffi::c_void = Box::into_raw(Box::new([0u8; 100])) as *mut _;
+                let out_length: *mut SmallInt = &mut 10;
+
+                #[allow(unused_mut, unused_assignments)]
+                let mut buffer_length: SmallInt = 0;
+                $(buffer_length = $buffer_length;)?
+
+                // Assert that the actual result matches expected
+                assert_eq!(
+                    $expected_sql_return,
+                    SQLGetInfoW(
+                        mongo_handle as *mut _,
+                        info_type,
+                        value_ptr as Pointer,
+                        buffer_length,
+                        out_length,
+                    )
+                );
+
+                // If the expectation is that the function returns successfully,
+                // assert that the resulting value and length are correct
+                match $expected_sql_return {
+                    SqlReturn::SUCCESS => {
+                        $(assert_eq!($expected_length, *out_length);)?
+                        $(assert_eq!($expected_value, $actual_value_modifier(value_ptr, *out_length as usize));)?
+                    },
+                    _ => ()
+                }
+
+                let _ = Box::from_raw(value_ptr);
+            }
+        }
+    }
+}
+
+unsafe fn modify_string_value(value_ptr: Pointer, out_length: usize) -> String {
+    input_wtext_to_string(value_ptr as *const _, out_length)
+}
+
+unsafe fn modify_u32_value(value_ptr: Pointer, _: usize) -> u32 {
+    *(value_ptr as *mut UInteger)
+}
+
+unsafe fn modify_u16_value(value_ptr: Pointer, _: usize) -> u16 {
+    *(value_ptr as *mut USmallInt)
+}
+
+mod unit {
+    use super::*;
+
+    test_get_info!(
+        driver_name,
+        info_type = InfoType::DriverName,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 40,
+        expected_length = 39,
+        expected_value = "MongoDB Atlas SQL interface ODBC Driver",
+        actual_value_modifier = modify_string_value,
+    );
+
+    test_get_info!(
+        driver_ver,
+        info_type = InfoType::DriverVer,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 11,
+        expected_length = 10,
+        expected_value = "00.01.0000",
```

Comment:
Yes we will need to change this whenever we release a new version. But that is useful! It will ensure we are actually returning the correct value whenever a new version is released. We could hypothetically calculate the version value at test time using the same code to calculate it in production, but then we wouldn't be testing anything really.